### DISCUSSION TITLE
`magick -list format` doesn't output anything in mingw build

### DISCUSSION THREAD

**Discussion by Biswa96** [ImageMagick/ImageMagick#5219] [Help] [answered]
In mingw, `magick -list format` does not show anything in the output. This happens only in mingw build. I have tried to debug the program but everything just went fine - no exception, no crashes. All the build options can be found [here](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-imagemagick/PKGBUILD). Can anyone provide any hint how to troubleshoot this issue?

**Comment by dlemstra** [comment, marked as answer]
My advise would be to patch the `ListMagickInfo` and put a bunch of `printf` statements in there to figure out what is happening?

**Reply by Biswa96** [reply]
Can you provide a little info about how magick gets list of formats? Does it read a file or search for some files in a directory?

**Reply by Biswa96** [reply]
Thanks for the hint. The main issue was a old patch with ImageMagick in msys2's mingw-packages repository. I have tracked the issue like this ListMagickInfo → GetMagickInfoList → GetMagickInfo → OpenModules → GetModuleList → NTOpenDirectory. In that old patch, the `L"\\*.*"` string was only `L"\\"` which caused the issue.

**Comment by snibgo** [comment]
Formats are defined in modules coders\XXX.c, for example coders\ftxt.c, function RegisterXXXImage(). The list of modules is in coders\Makefile.am and coders\coders_list.h.

**Reply by Biswa96** [reply]
Thank you too.

**Comment by Biswa96** [comment]
BTW, a little off-topic. There are some warnings with mingw build. Wondering if those can be fixed.

```
In file included from ../MagickCore/attribute.c:47:
../MagickCore/blob-private.h:31: warning: "fseek" redefined
   31 | # define fseek  fseeko
      |
In file included from ../MagickCore/studio.h:245,
                 from ../MagickCore/attribute.c:43:
../MagickCore/nt-base.h:264: note: this is the location of the previous definition
  264 | #    define fseek  _fseeki64
      |
../MagickCore/blob-private.h:32: warning: "ftell" redefined
   32 | # define ftell  ftello
      |
../MagickCore/nt-base.h:267: note: this is the location of the previous definition
  267 | #    define ftell  _ftelli64
      |
```

**Reply by dlemstra** [reply]
We are not getting those warnings in our own mingw builds: https://github.com/ImageMagick/ImageMagick/actions/runs/3215266232/jobs/5256247218. We are getting a bunch of others but I couldn't find a good fix for those.

**Reply by Biswa96** [reply]
Interesting! I get those warnings in ucrt64 which uses ucrt. In CI, mingw64 environment is used which depends on msvcrt. 🕵️
